includes/data-port/class-sensei-import-block-migrator.php

```diff
@@ -0,0 +1,234 @@
+<?php
+/**
+ * File containing the Sensei_Import_Course_Content_Migrator class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * This class is responsible for migrating post content which contains Sensei blocks.
+ */
+class Sensei_Import_Block_Migrator {
+
+	/**
+	 * The course id.
+	 *
+	 * @var int
+	 */
+	private $course_id;
+
+	/**
+	 * The data port task.
+	 *
+	 * @var Sensei_Data_Port_Task
+	 */
+	private $task;
+
+	/**
+	 * The course import model.
+	 *
+	 * @var Sensei_Import_Model
+	 */
+	private $import_model;
+
+	/**
+	 * Sensei_Import_Course_Content_Migrator constructor.
+	 *
+	 * @param int                   $course_id    The course which gets migrated.
+	 * @param Sensei_Data_Port_Task $task         The data port task which this migration is part of.
+	 * @param Sensei_Import_Model   $import_model The import model.
+	 */
+	public function __construct( int $course_id, Sensei_Data_Port_Task $task, Sensei_Import_Model $import_model ) {
+		$this->course_id    = $course_id;
+		$this->task         = $task;
+		$this->import_model = $import_model;
+	}
+
+	/**
+	 * Migrates the imported post content to use the ids of the newly created lessons and modules.
+	 *
+	 * @param string $post_content The post content.
+	 *
+	 * @return string The migrated post content.
+	 */
+	public function migrate( string $post_content ) : string {
+		if ( ! has_block( 'sensei-lms/course-outline', $post_content ) ) {
+			return $post_content;
+		}
+
+		$blocks = parse_blocks( $post_content );
+
+		$i = 0;
+		foreach ( $blocks as $block ) {
+			if ( 'sensei-lms/course-outline' === $block['blockName'] ) {
```

Comment:
The outline block is not always top-level, for example it could be wrapped in a Group block to give it a background. 
So we'll have to dig into the inner blocks here to find it.

Reply:
Great catch! 7f8b0d20460b49f5a6f26cadadda2e8e0c5bdbc2